parachain/primitives/router/src/outbound/mod.rs

```diff
@@ -132,7 +132,7 @@ where
 		})?;
 
 		// convert fee to MultiAsset
-		let fee = MultiAsset::from((MultiLocation::parent(), fee)).into();
+		let fee = MultiAsset::from((MultiLocation::parent(), fee.base_fee)).into();
```

Comment:
We should return `base_fee` instead of the `delivery_fee` for the  `XcmManager` to charge.

Reply:
We do need to make sure that the computed origin is charged the delivery fee. Otherwise the parachain that is the parent of the computed origin needs to estimate the fees themselves - this approach I am skeptical of.

In either case, lets wait until this PR is ready: https://github.com/paritytech/polkadot-sdk/pull/2021/files

It allows to flexibly decide how to handle the fee using a `HandleFee` trait.

Reply:
I do think it's the responsibility of source chain(i.e asset_hub or other sibling chain) to ensure all costs have been charged from their end user(signed origin).

More detail as we discussed before in https://snowfork.slack.com/archives/G01BGMGLAC9/p1696933144716239?thread_ts=1696931181.004119&cid=G01BGMGLAC9, the fee charged will be collected into treasury account of source chain and later use to rebalance their sovereign account on Ethereum.

But anyway, I'll stop here for the upstream PRs ready and we agree on the implementation.

Reply:
I think we can offer parachains the option to charge fees from their users themselves. For example, in future, the `create_channel` and `update_channel` extrinsics can give parachains flexibility in choosing the model they want.

However, by default, I want an easy way for parachains to integrate with our bridge, and charging fees on BridgeHub makes that possible.

Reply:
https://github.com/Snowfork/snowbridge/pull/975/commits/828063441d9ae5a2b0fbbdd331e608c646e09b78